src/util/popup.ts

```diff
@@ -19,31 +25,47 @@ export class PopupService<T> {
 
   constructor(
       private _type: any, private _injector: Injector, private _viewContainerRef: ViewContainerRef,
-      private _renderer: Renderer2, private _componentFactoryResolver: ComponentFactoryResolver,
-      private _applicationRef: ApplicationRef) {}
+      private _renderer: Renderer2, private _ngZone: NgZone,
+      private _componentFactoryResolver: ComponentFactoryResolver, private _applicationRef: ApplicationRef) {}
 
-  open(content?: string | TemplateRef<any>, context?: any): ComponentRef<T> {
+  open(content?: string | TemplateRef<any>, context?: any, animation = false):
+      {windowRef: ComponentRef<T>, transition$: Observable<void>} {
     if (!this._windowRef) {
       this._contentRef = this._getContentRef(content, context);
       this._windowRef = this._viewContainerRef.createComponent(
           this._componentFactoryResolver.resolveComponentFactory<T>(this._type), this._viewContainerRef.length,
           this._injector, this._contentRef.nodes);
     }
 
-    return this._windowRef;
-  }
+    const {nativeElement} = this._windowRef.location;
+    const onStable$ = this._ngZone.onStable.asObservable().pipe(take(1));
+    const transition$ = onStable$.pipe(
+        mergeMap(
+            () => ngbRunTransition(
+                nativeElement, ({classList}) => classList.add('show'), {animation, runningTransition: 'continue'})));
```

Comment:
To be discussed, but I'm not sure but here we can use `stop` instead of `continue`. I don't know if there is a reason to not be able to revert it

The question is : do we want to behave like the alert (not reversible) or like the dialog, toast or collapse (reversible) ?

Reply:
I would say reversible, as it works in two ways - opening/closing. I'll change that, thanks.